geoips/commandline/commandline_interface.py

```diff
@@ -83,9 +84,50 @@ def execute_command(self):
             sys.exit(2)
 
 
+def support_legacy_procflows():
+    """Run a series of checks on sys.argv to support legacy calls to run_procflow.
+
+    The new GeoIPS CLI has changed how we run our process workflows from now on. Instead
+    of calling 'run_procflow' or 'data_fusion_procflow', we now call
+    'geoips run single_source' or 'geoips run data_fusion'. This function parses through
+    sys.argv and performs the necessary translations to match our current format so that
+    the CLI's argparser can call the appropriate functionality.
+
+    Returns
+    -------
+    legacy: bool
+        - The truth value as to whether or not a legacy procflow call was used
+    """
+    defined_procflow = None
+    if "run_procflow" == basename(sys.argv[0]):
+        entrypoint = "run_procflow"
+        defined_procflow = "single_source"
```

Comment:
Does this also handle `config_based`?

Reply:
Oh, never mind. I see how this is done below. This could be more clear code-wise. Maybe we can have a separate function that determines the procflow rather than setting it, then re-setting it?

Reply:
The only reason it is implemented the way it is currently is that we don't currently enforce what procflows can be defined in the legacy ``--procflow <procflow_name>`` argument. By defaulting to either ``single_source`` or ``data_fusion``, we can ensure that a correct procflow is called, whether or not it works. @jsolbrig, we could check that if either ``run_procflow`` or ``data_fusion`` entry-points are in ``sys.argv``, and if ``--procflow`` is in ``sys.argv``, then we could enforce that it's one of ``config_based``, ``data_fusion``, or ``single_source``. Eventually we could just enforce ``order_based``.  What do you think?

Reply:
I'm thinking about this. It just occurred to me that this might not work if data_fusion isn't installed. Maybe we could test for a valid procflow by comparing against the available procflows? Do we have access to the available plugins at this point in the code?

Reply:
Discussed on a call:

- Will update such that when `geoips run <procflow> --legacy` is called, this will check the specified procflow against the list of installed procflows when running in legacy mode. If the procflow doesn't exist, an error will be raised.
- Will also update such that when `geoips run` is called without the `--legacy` option a `NotImpelemented` error will be raised. The error will indicate that, eventually, `geoips run` will always call the "order-based" procflow, but the order-based procflow is currently under development.

Reply:
See #598.